packages/dynamoose/lib/Table/index.ts

```diff
@@ -243,13 +243,23 @@ export class Table extends InternalPropertiesClass<TableInternalProperties> {
 					});
 				}, Promise.resolve());
 
-				await setupFlowPromise;
+				try {
+					await setupFlowPromise;
 
-				this.getInternalProperties(internalProperties).ready = true;
-				this.getInternalProperties(internalProperties).setupFlowRunning = false;
+					this.getInternalProperties(internalProperties).ready = true;
+					this.getInternalProperties(internalProperties).setupFlowRunning = false;
 
-				this.getInternalProperties(internalProperties).pendingTasks.forEach((task) => task());
-				this.getInternalProperties(internalProperties).pendingTasks = [];
+					this.getInternalProperties(internalProperties).pendingTasks.forEach((task) => task.resolve());
+					this.getInternalProperties(internalProperties).pendingTasks = [];
+				} catch (error) {
+					this.getInternalProperties(internalProperties).ready = false;
+					this.getInternalProperties(internalProperties).setupFlowRunning = false;
```

Comment:
Still not sure why these lines are required.

Reply:
If I really don't need it, I'll delete the line. @fishcharlie